### PR TITLE
Feature/create get api to fetch products with filters fh

### PR DESCRIPTION
**Pull request by hfatima08**

### CODE REVIEW

File: app/Http/Controllers/ProductController.php [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+            return $query->where('brand', $request->brand);
+        })
+        ->when($request->filled('minPrice') && $request->filled('maxPrice'), function ($query) use ($request) {
+            return $query->whereBetween('price', [$request->minPrice, $request->maxPrice]);
+        })
+        ->when($request->filled('keyword'), function ($query) use ($request) {
+            return $query->where('name', 'like', '%' . $request->keyword . '%');
+        })
+        ->get();
+
```
Comment (ZeeshanHabib7) [inline app/Http/Controllers/ProductController.php]: what if I want to fetch all the products ?

File: database/migrations/2024_07_02_121214_create_products_table.php [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+     */
+    public function up(): void
+    {
+        Schema::create('products', function (Blueprint $table) {
+            $table->increments('id');
+            $table->string('name');
+            $table->integer('price');
+            $table->date('date');
+            $table->string('brand');
+            $table->timestamps();
```
Comment (ZeeshanHabib7) [inline database/migrations/2024_07_02_121214_create_products_table.php:20]: add one more column 
  $table->softDeletes();

File: app/Models/Product.php [hunk 3]
```diff
@@ -4,9 +4,21 @@
...
+
+    public function getCreatedAtAttribute()
+    {
+        return Carbon::parse($this->attributes['created_at'])->format('Y-m-d H:i:s');
+    }
+
+    public function getUpdatedAtAttribute()
+    {
+        return Carbon::parse($this->attributes['updated_at'])->format('Y-m-d H:i:s');
+    }
```
Comment (ZeeshanHabib7) [inline app/Models/Product.php]: accessors are the good approach try to do it using Cast:
  protected $casts = [
        'created_at' => 'datetime:Y-m-d H:m:s',
        'updated_at' => 'datetime:Y-m-d H:m:s',
        'deleted_at' => 'datetime:Y-m-d H:m:s',
      ];

File: app/Http/Controllers/ProductController.php (hunk 1 above)
Comment (hfatima08) [reply app/Http/Controllers/ProductController.php]: http://127.0.0.1:8000/api/products using this route with no filters will fetch all the products

**Pull request closed without merging**